File: backend/UpWork/UpWork.Common/Models/DatabaseModels/AbsenceModel.cs
using System.Text.Json.Serialization;
using UpWork.Common.Enums;

namespace UpWork.Common.Models.DatabaseModels
{
    public class AbsenceModel
    {
        public Guid Id { get; set; }
        public DateTime FromDate { get; set; }
        public DateTime ToDate { get; set; }
        public bool IsActive { get; set; }
        //public Guid AbsenceTypeId { get; set; }
        //[JsonIgnore]
        //public virtual AbsenceTypeModel AbsenceType { get; set; }
        public AbsenceTypes AbsenceType { get; set; }
        public Guid UserId { get; set; }
        [JsonIgnore]
        public virtual UserModel User { get; set; }
        public ApprovalState ApprovalState { get; set; }
        public Guid TimeOffSupervisorId { get; set; }
        [JsonIgnore]
        public virtual UserModel TimeOffSupervisor { get; set; }
    }
}

File: frontend/upwork-client/src/app/models/absence.model.ts
import { AbsenceType } from './enums/absence-type.enum';
import { ApprovalState } from './enums/approval-state.enum';

export interface Absence {
  id?: string;
  fromDate: Date;
  toDate: Date;
  absenceType: AbsenceType;
  userId?: string;
  userFirstName?: string;
  userLastName?: string;
  approvalState: ApprovalState;
  timeOffSupervisorId?: string;
  supervisorFirstName?: string;
  supervisorLastName?: string;
}
